mobile/src/main/java/org/openhab/habdroid/ui/WriteTagActivity.java

```diff
@@ -139,76 +150,126 @@ public void onNewIntent(Intent intent) {
             return;
         }
 
-        Tag tagFromIntent = intent.getParcelableExtra(NfcAdapter.EXTRA_TAG);
-        Log.d(TAG, "NFC TAG = " + tagFromIntent.toString());
-        Log.d(TAG, "Writing page " + mSitemapPage + " to tag");
+        new AsyncTask<Void, Integer, Boolean>() {
+            @Override
+            protected Boolean doInBackground(Void... voids) {
+                Tag tag = intent.getParcelableExtra(NfcAdapter.EXTRA_TAG);
+                Log.d(TAG, "NFC TAG = " + tag.toString());
+                Log.d(TAG, "Writing page " + mSitemapPage + " to tag");
 
-        TextView writeTagMessage = findViewById(R.id.write_tag_message);
+                publishProgress(R.string.info_write_tag_progress);
```

Comment:
Given this is only called once the progress logic could/should be moved to onPreExecute.